jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,821 @@
+   "source": [
+    "#NIRSpec IFU science data cube\n",
+    "BoxPath=\"https://data.science.stsci.edu/redirect/JWST/jwst-data_analysis_tools/IFU_optimal_extraction/\"\n",
+    "filename=BoxPath+'NRS00001-faintQSO-F100LP-G140H-01_1_491_SE_2020-08-25T12h15m00_s3d.fits'\n",
+    "\n",
+    "#Open and inspect the file and WCS\n",
+    "#Load with astropy.fits.open\n",
+    "with fits.open(filename, memmap=False) as hdulist:\n",
+    "    sci = hdulist[\"SCI\"].data\n",
+    "    err = hdulist[\"ERR\"].data\n",
+    "    w = wcs.WCS(hdulist[1].header)\n",
+    "    hdr=hdulist[1].header\n",
+    "    hdulist.info()\n",
+    "    print(w)\n",
+    "    \n",
+    "#Load with Spectrum1D    \n",
+    "spec1d = Spectrum1D.read(filename)\n",
+    "\n",
+    "#Wavelengths\n",
+    "wavelength=np.array(spec1d.spectral_axis.value)\n",
+    "print(wavelength)\n",
+    "\n",
+    "#Sum over spaxels\n",
+    "fnu_sum=np.sum(spec1d.flux,axis=(0,1))\n",
+    "\n",
+    "#List of cube slices\n",
+    "data=[]\n",
+    "var=[]\n",
+    "spec1d_len=len(spec1d.spectral_axis.value)\n",
+    "for idx in range(spec1d_len): \n",
+    "    data.append(sci[idx,:,:])\n",
+    "    var.append(err[idx,:,:])   #variance=err, not variance=err**2.  Squaring the err gives strange, noisy results. \n",
+    "\n",
+    "#Window data, variance, and PSF model (and replace NaNs)\n",
+    "#The existing JWST pipeline window is overgenerous (39x33 instead of the nominal 30x30 pixels)\n",
+    "data_win=np.nan_to_num(np.array(data)[:,5:-4,3:])\n",
+    "data_var=np.nan_to_num(np.array(var)[:,5:-4,3:]) \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Can we fix or suppress this warning?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:*  Is there a way to read in with only Spectrum1D to perform all of our cube operations, not using fits.open()?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize Science Data with Cubeviz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Cubeviz incompatible with jupyter_client 6.1.6.  Use jupyter_client 5.3.5 instead."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from jdaviz import CubeViz\n",
+    "cubeviz = CubeViz()\n",
+    "cubeviz.app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### UI Instructions:\n",
+    "* Load science datacube into Cubeviz using the cell below\n",
+    "* Select a circular subset region centered on the source. \n",
+    "* Note that the region is pixelated and doesn't include fractional pixels\n",
+    "* Change the collapse method to 'Sum' in spectrum viewer: Gear icon : Viewer \n",
+    "* --This 'Sum' method yields our subset extraction\n",
```

Comment:
In the UI instructions indicate that after changing collapse method to Sum, the user needs to press the button "Apply" .
 I once ran the code *without* setting the method to Sum and I did not see any difference in the output. Could you please  explain why ? Is this necessary ?

Reply:
The default collapse method is 'Maximum', yielding a subset spectrum from the spaxel with maximum flux.  The 'Sum' method needs to be used to get the sum within the aperture.